framework/src/outputs/PNGOutput.C

```diff
@@ -0,0 +1,237 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#define nls _problem_ptr->getNonlinearSystem()
+#define scaledValue ((dv(0) + shiftValue) / scalingMax)
+#define reverseScale ((outBoundsControlColor * scalingMax) - shiftValue)
```

Comment:
Let's avoid the macros. None of them are necessary and they are context sensitive. Go ahead and create new variables or functions instead of using these.

Reply:
So I changed the scaledValue and reverseScale into inline functions, and I replaced nls macro symbol with the macro value everywhere it occurred. I kept running into problems with the other things I tried to do to create a variable that would end up giving the same value as the macro.